### PR TITLE
Fix determineState of ThermalHouse

### LINKED ISSUE #926: Fix determineState of ThermalHouse
When determining the state of ThermalHouse till the actual tick, the "old" ambientTemperature in the meaning temperature till the actual tick should be used. At the moment we use the ambientTemperature which can be updated within the actual tick.

This is neccessary for some issues described in #827

### PR DESCRIPTION
**Pull request by danielfeismann**
resolves #926

### CODE REVIEW

File: src/main/scala/edu/ie3/simona/model/thermal/ThermalGrid.scala [hunk 1]
```diff
@@ -115,19 +117,28 @@ final case class ThermalGrid(
...
 
   /** Handles the case, when a grid has infeed. First, heat up all the houses to
     * their maximum temperature, then fill up the storages
     * @param tick
     *   Current tick
+    * @param lastAmbientTemperature
+    *   Ambient temperature until this tick
     * @param ambientTemperature
-    *   Ambient temperature
+    *   Actual ambient temperature
```
Comment (sebastian-peter) [inline src/main/scala/edu/ie3/simona/model/thermal/ThermalGrid.scala:141]:
```suggestion
    *   Current ambient temperature
```

File: src/main/scala/edu/ie3/simona/model/thermal/ThermalGrid.scala [hunk 2]
```diff
@@ -115,19 +117,28 @@ final case class ThermalGrid(
...
+      state,
+      qDot,
+    )
 
   /** Handles the case, when a grid has infeed. First, heat up all the houses to
     * their maximum temperature, then fill up the storages
     * @param tick
     *   Current tick
+    * @param lastAmbientTemperature
+    *   Ambient temperature until this tick
```
Comment (sebastian-peter) [inline src/main/scala/edu/ie3/simona/model/thermal/ThermalGrid.scala:139]: Maybe to add some specificity:

```suggestion
    *   Ambient temperature valid up until (not including) the current tick
```

File: src/main/scala/edu/ie3/simona/model/thermal/ThermalHouse.scala [hunk 3]
```diff
@@ -298,19 +298,22 @@ final case class ThermalHouse(
...
-    *   current instance in time
+    *   Current instance in time
     * @param state
-    *   currently applicable state
+    *   Currently applicable state
+    * @param lastAmbientTemperature
+    *   Ambient temperature until this tick
     * @param ambientTemperature
-    *   Ambient temperature
+    *   Actual ambient temperature
```
Comment (sebastian-peter) [inline src/main/scala/edu/ie3/simona/model/thermal/ThermalHouse.scala:307]:
```suggestion
    *   Current ambient temperature
```

File: src/main/scala/edu/ie3/simona/model/thermal/ThermalGrid.scala [hunk 4]
```diff
@@ -248,8 +262,10 @@ final case class ThermalGrid(
     *
     * @param tick
     *   Current tick
+    * @param lastAmbientTemperature
+    *   Ambient temperature until this tick
```
Comment (sebastian-peter) [inline src/main/scala/edu/ie3/simona/model/thermal/ThermalGrid.scala:266]:
```suggestion
    *   Ambient temperature valid up until (not including) the current tick
```

File: src/main/scala/edu/ie3/simona/model/thermal/ThermalGrid.scala [hunk 5]
```diff
@@ -319,8 +338,10 @@ final case class ThermalGrid(
     *   Previous thermal house state before a first update was performed
     * @param formerStorageState
     *   Previous thermal storage state before a first update was performed
+    * @param lastAmbientTemperature
+    *   Ambient temperature until this tick
```
Comment (sebastian-peter) [inline src/main/scala/edu/ie3/simona/model/thermal/ThermalGrid.scala:342]:
```suggestion
    *   Ambient temperature valid up until (not including) the current tick
```

File: src/main/scala/edu/ie3/simona/model/thermal/ThermalHouse.scala [hunk 6]
```diff
@@ -298,19 +298,22 @@ final case class ThermalHouse(
   /** Update the current state of the house
     *
     * @param tick
-    *   current instance in time
+    *   Current instance in time
     * @param state
-    *   currently applicable state
+    *   Currently applicable state
+    * @param lastAmbientTemperature
+    *   Ambient temperature until this tick
```
Comment (sebastian-peter) [inline src/main/scala/edu/ie3/simona/model/thermal/ThermalHouse.scala:305]:
```suggestion
    *   Ambient temperature valid up until (not including) the current tick
```

File: src/main/scala/edu/ie3/simona/model/thermal/ThermalGrid.scala [hunk 7]
```diff
@@ -248,8 +262,10 @@ final case class ThermalGrid(
     *
     * @param tick
     *   Current tick
+    * @param lastAmbientTemperature
+    *   Ambient temperature until this tick
     * @param ambientTemperature
-    *   Ambient temperature
+    *   Actual ambient temperature
```
Comment (sebastian-peter) [inline src/main/scala/edu/ie3/simona/model/thermal/ThermalGrid.scala:268]:
```suggestion
    *   Current ambient temperature
```

File: src/main/scala/edu/ie3/simona/model/thermal/ThermalGrid.scala [hunk 8]
```diff
@@ -319,8 +338,10 @@ final case class ThermalGrid(
     *   Previous thermal house state before a first update was performed
     * @param formerStorageState
     *   Previous thermal storage state before a first update was performed
+    * @param lastAmbientTemperature
+    *   Ambient temperature until this tick
     * @param ambientTemperature
-    *   Ambient temperature
+    *   Actual ambient temperature
```
Comment (sebastian-peter) [inline src/main/scala/edu/ie3/simona/model/thermal/ThermalGrid.scala:344]:
```suggestion
    *   Current ambient temperature
```

File: src/main/scala/edu/ie3/simona/model/thermal/ThermalGrid.scala [hunk 9]
```diff
@@ -105,8 +105,10 @@ final case class ThermalGrid(
     *   Instance in time
     * @param state
     *   Currently applicable state
+    * @param lastAmbientTemperature
+    *   Ambient temperature until this tick
     * @param ambientTemperature
-    *   Ambient temperature
+    *   Actual ambient temperature
```
Comment (sebastian-peter) [inline src/main/scala/edu/ie3/simona/model/thermal/ThermalGrid.scala:111]:
```suggestion
    *   Current ambient temperature
```

File: src/main/scala/edu/ie3/simona/model/thermal/ThermalGrid.scala [hunk 10]
```diff
@@ -105,8 +105,10 @@ final case class ThermalGrid(
     *   Instance in time
     * @param state
     *   Currently applicable state
+    * @param lastAmbientTemperature
+    *   Ambient temperature until this tick
```
Comment (sebastian-peter) [inline src/main/scala/edu/ie3/simona/model/thermal/ThermalGrid.scala:109]:
```suggestion
    *   Ambient temperature valid up until (not including) the current tick
```

**Review by sebastian-peter — COMMENTED** [review]
Ok now I totally unterstand what you meant. Thanks for providing this fix!

Not sure if we need a test for this? Maybe testing `ThermalHouse#determineState`?

**Comment by danielfeismann** [comment]
> Ok now I totally unterstand what you meant. Thanks for providing this fix!
> 
> Not sure if we need a test for this? Maybe testing `ThermalHouse#determineState`?

Most probably: yes. I will add one.

**Review by sebastian-peter — APPROVED** [review]
Thanks!

**Pull request merged**